lib/src/util/utf8.cc

```diff
@@ -18,7 +19,15 @@ namespace Util {
                 return false;
             }
         }
-        return true;
+
+        // rapidjson::UTF8<char>::Validate accepts null bytes as valid UTF-8.
+        // They technically are valid since they're the null character. But
+        // null characters aren't valid in a string, so we want to disallow
+        // them regardless.
+        bool has_null = std::any_of(s.begin(), s.end(), [](char c) {
+                            return c == 0;
+                        });
+        return !has_null;
```

Comment:
Ah, so `std::any_of` will be false for an empty string http://www.cplusplus.com/reference/algorithm/any_of/  nice